src/components/legend/draw.js

```diff
@@ -170,10 +170,9 @@ module.exports = function draw(gd) {
 
             // Make sure the legend left and right sides are visible
             var legendWidth = opts._width,
-                legendWidthMax = gs.w;
+                legendWidthMax = gs.w * Math.max(1 - opts.x, 1);
```

Comment:
This calculation makes sense for the (default) `xanchor: 'left'` - seems like there's something similar for `'right'` like `* Math.max(opts.x, 1)`, not sure what it should look like for `'center'`, and `'auto'` should presumably depend on which alignment it auto-chose.

(and similarly with `legendHeightMax` and `yanchor`)